report.py

```diff
@@ -48,6 +50,35 @@ def get_translations(cls):
         return JinjaTranslator(lambda text: translate(text))
 
 
+class Jinja2ReportLogo(Jinja2Report):
+
+    @classmethod
+    def render(cls, action, context):
+        context['logo'] = cls.get_logo(context['user'])
+        return super().render(action, context)
+
+    @staticmethod
+    def get_logo_filename():
+        return 'logo.png'
+
+    @classmethod
+    def get_logo(cls, user):
+        Attachment = Pool().get('ir.attachment')
+        try:
+            attachment, = Attachment.search([
+                ('resource', '=', user.company),
+                ('name', '=', cls.get_logo_filename()),
+            ])
+            logo = (
+                'data:image/png;base64,'
+                + b64encode(attachment.data).decode()
```

Comment:
base64 encoding should only happen in the template if that is requried. Jinja2Report could be used to render documents other than html pages and a get_attachment utility function could be used to retrieve files other than images nor png images.